java/src/main/java/ai/rapids/cudf/Rmm.java

```diff
@@ -119,7 +119,25 @@ public static boolean isInitialized() throws RmmException {
   /**
    * Shut down any initialized rmm.
    */
-  public static native void shutdown() throws RmmException;
+  public static void shutdown() throws RmmException {
+    final int MAX_ATTEMPTS = 4;
+    int attempts = 0;
+    try {
+      while (!MemoryCleaner.bestEffortNoRmmBlockers() && attempts < MAX_ATTEMPTS) {
+        System.gc();
+        attempts++;
+        Thread.sleep(500);
```

Comment:
I'm a little worried about the case where RMM shutdown is being called as part of shutdown hooks and the system is under heavy load at the time.  Wondering if we should try a bit longer to try to handle that case more gracefully, otherwise I could see cases where this is thrown spuriously on a loaded VM (like during CI runs).  We could also poll more often (e.g.: every 10 milliseconds) to reduce unnecessary waiting when a GC cycle is needed.

Reply:
First of all I don't see many people shutting down RMM explicitly ever.  They could put it in as part of shutdown hooks, but there is no reason to, and it is most likely going to result in errors.  If the process goes away the memory is released.  I will update the documentation to make that clear, and explain the situations where they might want to shut it down.

So then there really are two questions.  How long should we wait for gc to finish and how much gc load to we want to put on the system.  I would say that we have some reasonable defaults and expose it to the end user so they have to decide.